### DISCUSSION TITLE
Best practices for nested Grids

### DISCUSSION THREAD

**Discussion by clhenrick** [carbon-design-system/carbon#9620] [Help] [answered]
Hi, after reviewing the guidelines and examples for the Carbon Grid system I did not find any information on best practices for implementing / using a nested grid (e.g. `.bx--grid > .bx--row > .bx--col > .bx--grid`). Is there any guidance or recommendations on when to do this, when to avoid it, etc.? Thanks!

**Comment by tay1orjones** [comment, marked as answer]
Hey @clhenrick 👋  Thanks for bringing this up - this is one shortfall of the current grid implementation. Nested grids unfortunately won't adhere to the parent grid's parameters. As such, typically columns in nested grids will need to use the various utility classes to add or remove margin/padding to ensure the columns align.

In the next major version of Carbon we are anticipating better support for nested grids as we add a new implementation based on CSS Grid. Using custom properties we've been able to provide correct column definitions in nested grids. It's currently in beta and a work in progress. So it's not fully documented and doesn't have much guidance surrounding it yet, but you can preview what we've been working on in [this storybook](https://carbon-react-next.netlify.app/?path=/docs/components-grid--wide#grid).
